src/routes/Products.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+import { useFetchProducts } from "../store/products/products.hooks";
+import { useEffect } from "react";
+import { Link } from "react-router-dom";
+
+const Products: React.FC = () => {
+  const { submit: fetchProducts, data = [], error } = useFetchProducts();
+
+  useEffect(() => {
+    fetchProducts();
+  }, []);
+
+  return (
+    <div>
+      <ul>
+        {data.map(product => (
```

Comment:
I can't quite see where the lower case `products` has come in? Is that not defined inside this script, but defined by the GET endpoint which we have obtained with `fetchProducts()` above?

Reply:
so `data` is defined on line 8, (it's [destructured](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) from the object return by the invocation of `useFetchProducts` hook.

`fetchProducts` is also defined in this way.

On line 11, `fetchProducts` is called. It is called only on the first render of the `Products` component, since the dependency array passed to `useEffect` (lines 10,11,12) is an empty list. When `fetchProducts` is called, api hooks stuff happens, which ultimately updates the value of the variable `data`.

`data` in this case is an array of products. When mapped over, we are naming the array element value which we are currently on `product`.

In JSX, in the return statement of the component function, we can map over arrays, and the return value of the map will be a list of components. These components are then rendered.

Reply:
So when we say 'product' here to refer to an element of an array (an element that happens to be a JSON object), we could actually have used an arbitrary string to refer to the element? But in contrast the array itself is given with the name 'products' on the endpoint api/products, so if we called it something other than products it would throw an error?

Reply:
Erm, no it could have any name.

```
interface Product {
  id: number
  name: string
}

const data: Product[] = [{
  id: 1,
  name: "Hella beer"
}, {
  id: 2,
  name: "Reid it and weap"
}]

const names = data.map(callthisvariableanything => callthisvariableanything.name);

console.log(names);

```

Paste that into here https://www.typescriptlang.org/play/# and click "Run"

Reply:
ok i see that cheers.